### PR TITLE
InMemory Implementation - Damian Modernell

### PR DESCRIPTION
**Pull request by damcho**

### CODE REVIEW

File: FeedStoreChallenge/MemoryFeedStore.swift [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+//
+//  Created by Damian Modernell on 18/3/2564 BE.
+//  Copyright © 2564 Essential Developer. All rights reserved.
+//
+
+import Foundation
+
+public final class MemoryFeedStore: FeedStore {
+	
+	private var feedCache: ([LocalFeedImage], Date) = ([], Date())
```
Comment (caiozullo) [inline FeedStoreChallenge/MemoryFeedStore.swift:13]: The default values are not needed - you can use an optional to represent the empty cache:

```suggestion
	private var feedCache: ([LocalFeedImage], Date)?
```

File: FeedStoreChallenge/MemoryFeedStore.swift [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+	}
+	
+	public func insert(_ feed: [LocalFeedImage], timestamp: Date, completion: @escaping InsertionCompletion) {
+		feedCache = (feed, timestamp)
+		completion(nil)
+	}
+	
+	public func retrieve(completion: @escaping RetrievalCompletion) {
+		let (cachedFeeed, timestamp) = (feedCache.0, feedCache.1)
+		if cachedFeeed.isEmpty {
```
Comment (caiozullo) [inline FeedStoreChallenge/MemoryFeedStore.swift:29]: An empty array of images is a valid cache. So the store shouldn't check for the contents of the array.

What determines if the cache is empty or not is the presence of a value there. You can make the cache optional and represent the empty state with `nil` (see my other comment).

**Review by caiozullo — COMMENTED** [review]
Hi @damcho, you almost passed the challenge!

Please, check my review comments and let me know once you're done so I can review it again. Thank you 🙂

File: FeedStoreChallenge/MemoryFeedStore.swift (hunk 2 above)
Comment (damcho) [reply FeedStoreChallenge/MemoryFeedStore.swift]: Thanks @caiozullo , ready for review

File: FeedStoreChallenge/MemoryFeedStore.swift (hunk 1 above)
Comment (damcho) [reply FeedStoreChallenge/MemoryFeedStore.swift]: 👍

File: FeedStoreChallenge/MemoryFeedStore.swift [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+//
+//  Created by Damian Modernell on 18/3/2564 BE.
+//  Copyright © 2564 Essential Developer. All rights reserved.
+//
+
+import Foundation
+
+public final class MemoryFeedStore: FeedStore {
+	
+	private var cache: (feed: [LocalFeedImage],timestamp:  Date)?
```
Comment (caiozullo) [inline FeedStoreChallenge/MemoryFeedStore.swift:13]: Fix spacing to be consistent with the rest of the code:

```suggestion
	private var cache: (feed: [LocalFeedImage], timestamp: Date)?
```

**Review by caiozullo — COMMENTED** [review]
Hi @damcho,

Congratulations, you've passed the challenge 🥇

- All tests are passing ✅ 
- You broke down the implementation into small commits ✅ 

I left one final comment. Feel free to push changes.

Great job!

**Pull request closed without merging**